src/genome.py

```diff
@@ -214,17 +368,32 @@ def disable_te(self, te: int) -> None:
         TEs are already inactive, so there is no need to do anything
         for those.
         """
-        ...  # FIXME
+        how_long = self.transposable_elements[te][1]
+        active_counter = 0
+        link = self.genome.head.next
+        for id, info in self.transposable_elements.items():
+            if id == te:
+                while info[0] != active_counter:
+                    link =  link.next
+                    active_counter += 1
+        
```

Comment:
Put a `break` after the `while` loop. There is no reason to look through the remaining TEs after you have processed the one you are interested in.